File: 0119_pascal_triangle_ii/get_row.go
package main

// Time: O(n**2)
// Space: O(n)
func getRow(rowIndex int) []int {
	res := []int{1}
	for i := 1; i < rowIndex+1; i++ {
		for j := len(res) - 1; j > 0; j-- {
			res[j] += res[j-1]
		}
		res = append(res, 1)
	}
	return res
}

func main() {

}

File: 0119_pascal_triangle_ii/get_row.py
from typing import List


def get_row(row_index: int) -> List[int]:
    """
    Time: O(n**2)
    Space: O(n)
    """
    res = [1]
    for i in range(1, row_index + 1):
        for j in range(len(res) - 1, 0, -1):
            res[j] = res[j - 1] + res[j]
        res.append(1)
    return res
